docs/arch/controller.md

```diff
@@ -0,0 +1,40 @@
+# Architecture Document for Controller Component
+
+```mermaid
+sequenceDiagram
+    actor User
+    User->>Controller: sends pipeline jobs as YAML and repository link
+    alt is request malformed
+        Controller-->>User: Nok
+    else is well
+        Controller-->>User: Ok
+    end
+    Controller->>Database: saves pipeline in database
+    loop over steps
+        Controller-->>Scheduler: sends step (over GRPC)
+        Scheduler-->>Controller: job succeeded or not
+    end
+    User->>Controller: get pipeline state
+```
+
+## Glossary
+- action : list of steps, a declarative file can only contain one action.
+- step : a step is the most atomic component of our pipeline.
+
+## Component Description
+- Controller component is an app directly interacting with the Monitor and the Scheduler.
```

Comment:
maybe describe its role in a sentence?

Reply:
like "a controller manages the order of execution of CI jobs (a.k.a. steps) and ensures they are properly executed" or smth like that